src/languageservice/services/yamlCompletion.ts

```diff
@@ -524,6 +524,24 @@ export class YamlCompletion {
 
     this.finalizeParentCompletion(result);
 
+    const uniqueItems = result.items.filter(
```

Comment:
Hello, I have a small note here, not sure if I should create an issue for that...

but this piece of code (and test modifications below) is a little bit hiding another problem that some internal methods are called multiple times from different places with the same result.

for example, some schema.snippets are calculated multiple times

A user can't see any problems (results are merged), but it's not correct.

If you decide to create a task for that, I can try to solve it.

Thanks for your great open-source project!!